src/peer_discovery.cpp

```diff
@@ -151,6 +158,42 @@ PeerDiscovery::DomainPeerDiscovery::loopListener()
     });
 }
 
+
+void
+PeerDiscovery::DomainPeerDiscovery::query(const asio::ip::udp::endpoint peer)
+{
+    std::lock_guard<std::mutex> lck(mtx_);
+    if (not lrunning_)
+        return;
+
+    // array<string, 1> qitems = { "id" };
+    /*
+    struct request {
+        char *t;
+        items;
+    } struc_request = {
+        .t = "q";
+        .items = qitems;
+    };
+    */
+
+    msgpack::sbuffer pbuf_request;
+    std::map<std::string, std::string> request;
+    request["q"] = "id";
```

Comment:
why this format ?
Maybe instead of `{"q": "id"}`
It could be just `"q"`:
```c++
msgpack::sbuffer pbuf_request;
msgpack::pack(pbuf_request, "q");
```